docs/source/data/subscriptions.mdx

```diff
@@ -40,20 +40,108 @@ subscription PostFeed {
 > Each subscription operation can subscribe to only _one_ field of the `Subscription` type.
 
-## Setting a subscription endpoint
+## Introducing subscriptions to your Apollo Server
 
-Because subscriptions use WebSocket instead of HTTP, Apollo Server uses a _second_ GraphQL endpoint specifically for subscriptions. This endpoint uses the `ws` protocol instead of `http`.
+> Subscriptions are **not** supported by the "batteries-included" `apollo-server` package. In order to implement subscriptions, you must first "eject" (TODO: link to ejecting docs) and use the `apollo-server-express` package (or any other Apollo Server integration package which supports subscriptions). The following steps assume you've already ejected and are using `apollo-server-express`.
 
-By default, the subscription endpoint's path matches the path of your primary GraphQL endpoint (`/graphql` if not set). You can specify a different path for your subscription endpoint like so:
+In order to run both an Express app and a separate subscription server, we're going to create an `http.Server` instance which will effectively wrap the two and become our new `listen`er. We will also create a `SubscriptionServer` which will require a few tweaks in our existing code.
+
+1. Install `subscriptions-transport-ws` and `@graphql-tools/schema`.
+    ```sh
+    npm install subscriptions-transport-ws @graphql-tools/schema
+    ```
+
+1. Add the following imports in the module where your Apollo Server is currently instantiated. We'll use these in the subsequent steps.
+    ```javascript
+    import { createServer } from 'http';
+    import { execute, subscribe } from 'graphql';
+    import { SubscriptionServer } from 'subscriptions-transport-ws';
+    import { makeExecutableSchema } from '@graphql-tools/schema';
+    ```
+
+1. Create an `http.Server` instance with your Express `app`.
+
+    In order to set up both the HTTP and WebSocket servers, we'll need to create an `http.Server`. Do this by passing your Express `app` to the `createServer` function which we imported from the Node.js `http` module.
+
+    ```javascript
+    // This `app` is the returned value from `express()`.
+    const httpServer = createServer(app);
+    ```
+
+1. Create an instance of `GraphQLSchema` (if one doesn't already exist).
+    > Your server may already pass a `schema` to the `ApolloServer` constructor. If it does, this step can be skipped. You'll use the existing `schema` instance in a later step.
+
+    The `SubscriptionServer` (which we'll instantiate next) doesn't accept `typeDefs` and `resolvers` directly, but rather an executable `GraphQLSchema`. We can pass this `schema` object to both the `SubscriptionServer` and `ApolloServer`. This way, it's clear that the same schema is being used in both places.
+
+    ```javascript
+    const schema = makeExecutableSchema({ typeDefs, resolvers });
+    // ...
+    const server = new ApolloServer({
+      schema,
+    });
+    ```
+1. Create the `SubscriptionServer`.
```

Comment:
Hmm.

How about instead of doing nothing with the return value, we set up some signal handlers, here and in the migration guide?

```js
['SIGINT', 'SIGTERM'].forEach(signal => process.on(signal, () => subscriptionServer.close())
```

we can still link to #5074 in a command but just saying that shutdown ordering would be better if #5074 is fixed